src/tag/tag.service.ts

```diff
@@ -5,7 +5,19 @@ import { DbService } from '../db/db.service';
 export class TagService {
   constructor(private readonly prisma: DbService) {}
 
-  async getAllTags() {
-    return await this.prisma.tag.findMany();
+  async searchTags(search: string) {
+    const exactMatch = await this.prisma.tag.findFirst({
+      where: { name: search },
+    });
+    const partialMatch = await this.prisma.tag.findMany({
+      where: { name: { contains: search } },
+      orderBy: { name: 'asc' },
+    });
+    return {
+      tags: [...(exactMatch ? [exactMatch] : []), ...partialMatch].filter(
+        (tag, index, self) =>
+          self.findIndex((t) => t.name === tag.name) === index,
```

Comment:
It is unclear for me what is this code responsible for

Reply:
This code is responsible for setting the first element to `exactMatch`.

Reply:
What about stuff which happens in filter method

Reply:
Deleting duplicates values (exactMatch, if exists will also be in partialMatch array)

Reply:
I see you never heard about something like Set in JS
You can just transform Array into Set, then Set into Array and duplicates are gone while code remains clean

Reply:
Sets won't work there, because we are removing duplicates from array of objects.